WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domain/NewSiteCreationDomainViewHolder.kt

```diff
@@ -0,0 +1,84 @@
+package org.wordpress.android.ui.sitecreation.domain
+
+import android.support.annotation.LayoutRes
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RadioButton
+import android.widget.TextView
+import org.wordpress.android.R
+import org.wordpress.android.R.color
+import org.wordpress.android.R.drawable
+import org.wordpress.android.ui.sitecreation.domain.NewSiteCreationDomainsViewModel.DomainsListItemUiState
+import org.wordpress.android.ui.sitecreation.domain.NewSiteCreationDomainsViewModel.DomainsListItemUiState.DomainsFetchSuggestionsErrorUiState
+import org.wordpress.android.ui.sitecreation.domain.NewSiteCreationDomainsViewModel.DomainsListItemUiState.DomainsModelUiState
+
+sealed class NewSiteCreationDomainViewHolder(internal val parent: ViewGroup, @LayoutRes layout: Int) :
+        RecyclerView.ViewHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false)) {
+    abstract fun onBind(uiState: DomainsListItemUiState)
+
+    class DomainSuggestionItemViewHolder(
+        parentView: ViewGroup
+    ) : NewSiteCreationDomainViewHolder(parentView, R.layout.new_site_creation_domains_item) {
+        private val container = itemView.findViewById<ViewGroup>(R.id.container)
+        private val suggestion = itemView.findViewById<RadioButton>(R.id.domain_suggestion)
+        private val divider = itemView.findViewById<View>(R.id.divider)
+        private var onDomainSelected: (() -> Unit)? = null
+
+        init {
+            suggestion.buttonTintList = ContextCompat.getColorStateList(
+                    parentView.context,
+                    R.color.grey_blue_radio_button_state_list
+            )
+            // We need to trigger `onDomainSelected` for both the radio button and the container because otherwise
+            // some parts of the layout will not be clickable.
+            container.setOnClickListener {
```

Comment:
It was the bottom of the row, I'll try to reproduce and record a gif for you.

Reply:
On second thought, this was not exactly what I tried, so I tried your solution and it seemed to have worked great. Thank you! Here is the change: bb6443c1ac3d963b8e95a75126e2267f5179a5df. The only difference is we don't set `isChecked` manually because `onDomainSelected` will do it for us. I wasn't sure whether it's better to set `isChecked` first and then trigger `onDomainSelected` or not set it at all, and went with the latter. Let me know if you prefer otherwise.